docs/tutorials/hpa-recommendation.md

```diff
@@ -0,0 +1,219 @@
+    status: "True"
+    type: Ready
+  lastUpdateTime: "2022-05-15T13:34:19Z"
+  recommendedValue: |
+    maxReplicas: 2
+    metrics:
+    - resource:
+        name: cpu
+        target:
+          averageUtilization: 75
+          type: Utilization
+      type: Resource
+    minReplicas: 2
+```
+
+## HPA Recommendation Algorithm model
+
+### Inspecting
+
+1. Workload with low replicas: If the replicas is too low,  may not be suitable for hpa recommendation. Associated configuration: ehpa.deployment-min-replicas | ehpa.statefulset-min-replicas | ehpa.workload-min-replicas
+2. Workload with a certain percentage of not running pods: if the workload of Pod mostly can't run normally, may not be suitable for flexibility. Associated configuration: ehpa.pod-min-ready-seconds | ehpa.pod-available-ratio
+3. Workload with low cpu usage: The low CPU usage workload means that there is no load pressure. In this case, we can't estimate it. Associated configuration: ehpa.min-cpu-usage-threshold
+4. Workload with low fluctuation of cpu usage: dividing of the maximum and minimum usage is defined as the fluctuation rate. If the fluctuation rate is too low, the workload will not benefit much from hpa. Associated configuration: ehpa.fluctuation-threshold 
+
+### Advising
+
+In the advising phase, one EffectiveHPA Spec is recommended using the following Algorithm model. The recommended logic for each field is as follows:
+
+**Recommend TargetUtilization**
+
+Principle: Use Pod P99 resource utilization to recommend hpa. Because if the application can accept this utilization over P99 time, it can be inferred as a target for elasticity.
+
+1. Get the Pod P99 usage of the past seven days by Percentile algorithm: pod_cpu_usage_p99
+2. Corresponding utilization: target_pod_CPU_utilization = pod_cpu_usage_p99 / pod_cpu_request
+3. To prevent over-utilization or under-utilization, target_pod_cpu_utilization needs to be less than ehpa.min-cpu-target-utilization and greater than ehpa. max-cpu-target-utilization
+
+**Recommend minReplicas**
+
+Principle: MinReplicas are recommended for the lowest hourly workload utilization for the past seven days.
+
+1. Calculate the lowest median workload cpu usage of the past seven days: workload_cpu_usage_medium_min
+2. Corresponding replicas: minReplicas = workload_cpu_usage_medium_min / pod_cpu_request / ehpa.max-cpu-target-utilization
+3. To prevent the minReplicas being too small, the minReplicas must be greater than or equal to ehpa.default-min-replicas
+
+**Recommend maxReplicas**
+
+Principle: Use workload's past and future seven days load to recommend maximum replicas.
+
+1. Calculate P95 workload CPU usage for the past seven days and the next seven days: workload_cpu_usage_p95
+2. Corresponding replicas: max_replicas_origin = workload_cpu_usage_p95 / pod_cpu_request / target_cpu_utilization
+3. To handle with the peak traffic, Magnify by a certain factor: max_replicas = max_replicas_origin * ehpa.max-replicas-factor
```

Comment:
If the calculation statement renders by LaTeX.
It's better than inline paste text.
![](https://latex.codecogs.com/svg.image?%5Cinline%20%5Chuge%20target%5C_pod%5C_CPU%5C_utilization%20=%20%20%5Cfrac%7Bpod%5C_cpu%5C_usage%5C_p99%7D%7Bpod%5C_cpu%5C_request%7D%20)

---

I will refactor that after pr merge.